src/components/MAPI/apps/ClusterDetailAppListWidgetUninstall.tsx

```diff
@@ -0,0 +1,144 @@
+
+    try {
+      await deleteAppWithName(
+        appClient,
+        auth,
+        app.metadata.namespace!,
+        app.metadata.name
+      );
+
+      mutate(
+        applicationv1alpha1.getAppListKey({ namespace: app.metadata.namespace })
+      );
+
+      if (onAppUninstalled) onAppUninstalled();
+
+      new FlashMessage(
+        `App <code>${app.metadata.name}</code> will be uninstalled from cluster <code>${app.metadata.namespace}</code>.`,
+        messageType.SUCCESS,
+        messageTTL.SHORT
+      );
+    } catch (err) {
+      ErrorReporter.getInstance().notify(err as Error);
+
+      const errorMessage = extractErrorMessage(err);
+      new FlashMessage(
+        'Something went wrong while trying to uninstall your app.',
+        messageType.ERROR,
+        messageTTL.LONG,
+        errorMessage
+      );
+    }
+  };
+
+  const [isConfirmationVisible, setIsConfirmationVisible] = useState(false);
+  const handleCancel = () => setIsConfirmationVisible(false);
+  const handleUninstall = () => setIsConfirmationVisible(true);
+
+  return (
+    <ClusterDetailAppListWidget
+      title='Uninstall'
+      contentProps={{
+        margin: { top: 'small' },
+        pad: 'medium',
+        background: 'background-contrast',
+        round: { size: 'xsmall' },
+      }}
+      {...props}
+    >
+      {!app ? (
+        <Box direction='row' justify='between' align='center'>
+          <LoadingPlaceholder width={500} height={20} />
+          <LoadingPlaceholder width={100} height={36} />
+        </Box>
+      ) : (
+        <>
+          <ConfirmationPrompt
+            title={
+              <Text weight='bold' margin={{ bottom: 'small' }}>
+                Are you sure you want to uninstall {app.metadata.name} from
+                cluster{' '}
+                <ClusterIDLabel
+                  clusterID={app.metadata.namespace!}
+                  variant={ClusterIDLabelType.Name}
+                />
+                ?
+              </Text>
+            }
+            confirmButton={
+              <Button danger onClick={uninstallApp}>
+                Uninstall
+              </Button>
+            }
+            onConfirm={uninstallApp}
+            onCancel={handleCancel}
+            open={isConfirmationVisible}
+            contentProps={{
+              pad: { left: '0', top: 'xsmall' },
+              background: 'transparent',
+            }}
+          >
+            <Text>There is no undo.</Text>
+          </ConfirmationPrompt>
+
+          {!isConfirmationVisible && (
+            <Box
+              direction='row'
+              justify='between'
+              align='center'
+              animation={{ type: 'fadeIn', duration: 300 }}
+            >
+              <Text>
+                Uninstalling the app from this cluster will not remove the
+                configuration resources.
+              </Text>
+              <Button secondary onClick={handleUninstall}>
+                Uninstall...
```

Comment:
This is the first thing that Marian will see 😆 We use the UTF-8 ellipsis character in other places.
```suggestion
                Uninstall…
```